Sources/SwiftParser/Expressions.swift

```diff
@@ -78,6 +78,10 @@ extension Parser {
     ///
     /// We don't allow allow newlines here.
     case poundIfDirective
+
+    /// Parsing an attribute argument, which can contain declaration references
+    /// like `subscript` or `deinit`.
+    case attributeArgument
```

Comment:
Nit, this is parsing multiple arguments, so I would make it plural

```suggestion
    case attributeArguments
```

Reply:
Sure